middleware/pkg/tls/tls.go

```diff
@@ -22,5 +49,5 @@ func NewTLSConfig(certPath, keyPath, caPath string) (*tls.Config, error) {
 	if !ok {
 		return nil, fmt.Errorf("Could not read root certs: %s", err)
 	}
-	return &tls.Config{Certificates: []tls.Certificate{cert}, RootCAs: roots}, nil
```

Comment:
I'll file a bug to we should create a tls package to put this stuff in to aid in reuse for this and etcd middleware.